src/app/components/clients/clients.component.ts

```diff
@@ -18,70 +16,66 @@ import {SubSink} from 'subsink';
 
 export class ClientsComponent implements OnInit {
   realm: Realm;
-  client: Client;
-  clients: Client[];
+  clients;
   displayedColumns: string[] = ['name'];
   subSink = new SubSink();
-
-  form = new FormGroup({
-    clientName: new FormControl('', Validators.required),
-  });
+  edit = false;
 
   constructor(private changeDetectorRefs: ChangeDetectorRef,
               private service: ClientService,
-              private snackBar: SnackBarService,
               private dialog: MatDialog,
               private clientService: ClientService,
               private realmService: RealmService) {
   }
 
   ngOnInit(): void {
-    this.subSink.add(this.realmService.realm.subscribe((data: ParentRealm) => {
-      this.clients = data.clients;
-      this.realm = data.realm;
+    this.subSink.add(this.realmService.realm$.subscribe((data: Realm) => {
+      this.realm = data;
+      this.getAllClients();
     }));
   }
 
+  getAllClients() {
+    this.subSink.add(this.clientService.getAllClients(this.realm.name).subscribe((clients: Client[]) => this.clients = clients));
+  }
+
   ngOnDestroy() {
     this.subSink.unsubscribe();
   }
 
-  updateClient(currentClientName: string) {
-    const dialogRef = this.dialog.open(ClientDialogComponent);
+  updateClient(client) {
+    const dialogRef = this.dialog.open(ClientDialogComponent, {
+      data: {edit: this.edit}
+    });
```

Comment:
trimite direct true / false in functie de ce modal vrei, nu e nevoie sa setezi o variabila pentru asta